### PR TITLE
Error summary

### PR DESCRIPTION
**Pull request by jacobwindsor**
Adds error summary and also simplifies PaginatedDataSource and MetadataDataSource a bit

### CODE REVIEW

File: client/src/app/submission/submission.component.html [hunk 1]
```diff
@@ -28,6 +28,39 @@ <h2 class="vf-text vf-text-heading--2">
...
+        </li>
+        <li>
+          Protocols:
+          <a href="javascript: void(0)"(click)="navigateToTab(2, { dataSource: protocolsDataSource, filterState: 'Invalid' })">
+            {{validationSummary.invalidProtocols}} error(s)
+          </a>
+        </li>
+        <li>
+          Data files:
+          <a href="javascript: void(0)" (click)="navigateToTab(3, { dataSource: filesDataSource, filterState: 'Invalid'})">
```
Comment (aaclan-ebi) [inline client/src/app/submission/submission.component.html:55]: I think this should be 'Invalid File'

File: client/src/app/submission/submission.component.html [hunk 2]
```diff
@@ -28,6 +28,39 @@ <h2 class="vf-text vf-text-heading--2">
   <div>
     <div class="vf-text-body vf-text-body--2">{{projectTitle}}</div>
     <div class="vf-text-body vf-text-body--3">{{getContributors(project)}}</div>
+    <div *ngIf="validationSummary">
+      <p>Your metadata validation returned <span class="error-text">{{validationSummary.totalInvalid}} error(s)</span>. Review and fix them below.</p>
```
Comment (aaclan-ebi) [inline client/src/app/submission/submission.component.html:32]: remove 'metadata', as totalInvalid may include both data and metadata validation.

**Review by aaclan-ebi — APPROVED** [review]
I tested this locally, it looks great!

File: client/src/app/submission/submission.component.html (hunk 1 above)
Comment (jacobwindsor) [reply client/src/app/submission/submission.component.html:55]: This raises another issue. In fact, I think we would need "Invalid File", "Invalid Metadata", and "Missing File" right? The summary endpoint only returns counts for invalidFiles and missingFiles

Comment (jacobwindsor) [reply client/src/app/submission/submission.component.html:55]: @prabh-t is adding these properties to the summary endpoint. I will check on this with @gabsie later on and implement per her specs

**Pull request merged**